.github/workflows/prepare_release.yml

```diff
@@ -0,0 +1,58 @@
+---
+
+name: Prepare Release
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        description: 'Module version to be released.'
+        required: true
+        type: string
+      working-directory:
+        description: The working directory where all jobs should be executed. Used for modules in subdirectories like a monorepo or a control repository.
+        default: '.'
+        required: false
+        type: string
+
+env:
+  BUNDLE_WITHOUT: development:test:system_tests
+  GIT_AUTHOR_NAME: Release Automation
+  GIT_AUTHOR_EMAIL: "${{ github.repository_owner }}@users.noreply.github.com"
+
+jobs:
+  prepare_release:
+    defaults:
+      run:
+        working-directory: ${{ inputs.working-directory }}
+    name: 'Puppet Forge'
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.3'
+          bundler-cache: true
+          working-directory: ${{ inputs.working-directory }}
+      - name: Update metadata.json to new version
+        run: bundle exec rake module:bump_to_version[${{ inputs.version }}]
+      - name: Update CHANGELOG.md and update REFERENCE.md if required
+        env:
+          # https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication
+          CHANGELOG_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: bundle exec rake release:prepare
+      - name: 'Check if a release is necessary'
+        id: 'check'
+        run: |
+          git diff --quiet CHANGELOG.md && echo "release=false" >> $GITHUB_OUTPUT || echo "release=true" >> $GITHUB_OUTPUT
+      - name: 'Commit changes'
+        if: ${{ steps.check.outputs.release == 'true' }}
+        env:
+          # https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git switch --create release_prep
+          git commit --all --message="Release ${{ steps.get_version.outputs.version }}"
+          gh pr create --title "Release ${{ steps.get_version.outputs.version }}" --label skip-changelog
```

Comment:
How does `gh pr create` work if a PR already exists? Does it at least push to the existing branch or just error out? You know how common it's to create a prep, fix labels and then regenerate it.

Reply:
I don't know yet. That's why I raised https://github.com/voxpupuli/puppet-example/pull/81 , to test it.

Reply:
when running it on code where a PR already exists:

```
$ gh pr create
a pull request for branch "prep" into branch "v3" already exists:
https://github.com/voxpupuli/gha-puppet/pull/66
```